Documentation/project-docs/eventpipe.md

```diff
@@ -0,0 +1,106 @@
+Using EventPipe
+===============
+
+# Capabilities #
+EventPipe provides the ability to collect traces using an in-process event logger instead of depending on a platform-level logger.  The following data can be collected:
+
+ - Sampled thread-time traces (Samples of every thread collected on a recurring basis at the specified (or default) rate)
+ - Runtime events (Events emitted by the native components of the runtime - e.g. GC, JIT, Interop, etc.)
+ - EventSource events
+ - Managed call stacks for all events.
+
+# Collection Examples #
+
+## Start/Stop Collection at Any Time with Default Collection Parameters ##
+
+1. Tell the runtime to listen for enable/disable commands:
+
+    > ```cmd
+    > export COMPlus_EnableEventPipe=4
+    > ```
+
+2. Specify the output file path:
+
+    > ```cmd
+    > export COMPlus_EventPipeOutputFile=/path/to/trace.netperf
+    > ```
+
+3. Run the application.
+
+4. Start collection by creating an empty control file.  This tells the runtime to start collection:
+
+    > ```cmd
+    > touch /path/to/trace.netperf.ctl.
+    > ```
+
+    NOTE: The runtime polls every 10 seconds, so it can take up to 10 seconds for tracing to start.  You'll see the netperf file get created when tracing has started.
+
+5. Reproduce the behavior you want to capture.
+
+6. Stop collection by deleting the control file.  This tells the runtime to stop collection:
+
+    > ```cmd
+    > rm /path/to/trace.netperf.ctl
+    > ```
+
+This produces /path/to/trace.netperf which can be consumed using [TraceEvent and PerfView](http://github.com/microsoft/perfview).
+
+## Collect a Heap Dump ##
+
+1. Tell the runtime to listen for enable/disable commands:
+
+    > ```cmd
+    > export COMPlus_EnableEventPipe=4
+    > ```
+
+2. Specify the output file path:
+
+    > ```cmd
+    > export COMPlus_EventPipeOutputFile=/path/to/trace.netperf
+    > ```
+
+3. Specify the following collection configuration, which contains the default configuration + heap dump collection:
+
+    > ```cmd
+    > export COMPlus_EventPipeConfig=Microsoft-Windows-DotNETRuntime:0x4C1DFCCBD:5,Microsoft-Windows-DotNETRuntimePrivate:0x4002000B:5,Microsoft-DotNETCore-SampleProfiler:0x0:5
```

Comment:
You should not need private provider or the sample profiling provider to do a heap dump.   
You SHOULD just need the keyword 0x1180109 at informational (4) level.   (Please test it).    See https://github.com/Microsoft/perfview/blob/master/src/EtwHeapDump/DotNetHeapDumper.cs#L47

We should consider making shorcuts for common configurations.   For example we could make $ be special config name marker and 
```
export COMPlus_EventPipeConfig=$HeapDump
```
means 
```
export COMPlus_EventPipeConfig=Microsoft-Windows-DotNETRuntime:0x1180109
```
What I want is for it to be at least conceivable that you could do this without a script or cut/paste.

Reply:
Alternatively you could make the number used for COMPlus_EnableEventPipe indicate the default config.

Reply:
Let me think about how best to do this.  I suspect ```COMPlus_EventPipeConfig=$HeapDump``` would probably be the right way.  We can also have $GCCollect and $GCCollectOnly as well.